src/lightning/pytorch/trainer/connectors/data_connector.py

```diff
@@ -333,83 +318,80 @@ def _reset_eval_dataloader(
         Returns:
             Tuple (num_batches, dataloaders)
         """
-        # always get the loaders first so we can count how many there are
         dataloaders = self._request_dataloader()
+        if not isinstance(dataloaders, CombinedLoader):
+            combined_loader = CombinedLoader(dataloaders, "sequential")
+        else:
+            combined_loader = dataloaders
 
-        if self.trainer.overfit_batches > 0:
-            dataloaders = self._resolve_overfit_batches(dataloaders, mode)
+        trainer = self.trainer
 
-        # TODO(carmocca): list conversion shouldn't be forced
-        if not isinstance(dataloaders, list):
-            dataloaders = [dataloaders]  # type: ignore[assignment]
+        if trainer.overfit_batches > 0:
+            self._resolve_overfit_batches(combined_loader, mode)
 
-        if any(dl is None for dl in dataloaders):
-            rank_zero_warn("One of given dataloaders is None and it will be skipped.")
+        loader_num_batches: List[Union[int, float]] = []
+        module = model or trainer.lightning_module or self.datamodule
 
-        for loader in dataloaders:
-            apply_to_collection(
-                loader.iterables if isinstance(loader, CombinedLoader) else loader,
-                DataLoader,
-                self._check_eval_shuffling,
-                mode=mode,
+        dataloaders = []
+        for i, dl in enumerate(combined_loader.flattened):
+            is_shuffled = _is_dataloader_shuffled(dl)
+            # limit this warning only for samplers assigned automatically when shuffle is set
+            if is_shuffled:
+                rank_zero_warn(
+                    f"Your `{mode.dataloader_prefix}_dataloader`'s sampler has shuffling enabled,"
+                    " it is strongly recommended that you turn shuffling off for val/test/predict dataloaders.",
+                    category=PossibleUserWarning,
+                )
```

Comment:
This needs to compute `is_shuffled` before for so that we pass it to `_prepare_dataloader`, so what was `_check_eval_shuffling` before would now just be a wrapper for the `rank_zero_warn` (it would also compute `is_shuffled` before). Which I thought was unnecessary to have separate as a staticmethod.